plugins/org.eclipse.fordiac.ide.library/src/org/eclipse/fordiac/ide/library/LibraryManager.java

```diff
@@ -498,13 +498,14 @@ private static void createTypeEntriesManually(final IFolder folder, final Map<St
 		final IResourceVisitor visitor = resource -> (switch (resource) {
 		case final IFolder f -> true;
 		case final IFile file -> {
-			final TypeEntry entry = TypeEntryFactory.INSTANCE.createTypeEntry(file);
-			final TypeEntry oldEntry = cachedTypes.get(entry.getFullTypeName());
-			if (oldEntry != null) {
-				FordiacResourceChangeListener.updateTypeEntry(file, oldEntry);
-				cachedTypes.remove(entry.getFullTypeName());
-			} else {
-				typeLibrary.createTypeEntry(file);
+			if (TypeEntryFactory.INSTANCE.createTypeEntry(file) instanceof final TypeEntry entry) {
```

Comment:
I would find a null check clearer here. The return type is always a TypeEntry.

Reply:
yes, a null check is clearer